query/src/main/scala/filodb/query/exec/MetadataRemoteExec.scala

```diff
@@ -60,11 +63,30 @@ case class MetadataRemoteExec(queryEndpoint: String,
   def toQueryResponse(response: MetadataSuccessResponse, id: String, parentSpan: kamon.trace.Span): QueryResponse = {
       if (response.data.isEmpty) mapTypeQueryResponse(response, id)
       else response.data.head match {
-        case _: MetadataSampl => mapTypeQueryResponse(response, id)
-        case _ => labelsQueryResponse(response, id)
+        case _: MetadataSampl            => mapTypeQueryResponse(response, id)
+        case _: LabelCardinalitySampl    => mapLabelCardinalityResponse(response, id)
+        case _                           => labelsQueryResponse(response, id)
       }
   }
 
+  private def mapLabelCardinalityResponse(response: MetadataSuccessResponse, id: String): QueryResponse = {
+
+    import NoCloseCursor._
+    val data = response.data.asInstanceOf[Seq[LabelCardinalitySampl]]
+      .map(lc => {
+          val key = CustomRangeVectorKey(Map(
+            "_ws_".utf8 -> lc.ws.utf8,
+            "_ns_".utf8 -> lc.ns.utf8,
+            "_metric_".utf8 -> lc.metric.utf8))
+          val data = Seq(lc.cardinality.map(k => (k.getOrElse(lcLabelNameField, "").utf8,
```

Comment:
Worth logging an `info` if the "else" is followed?

Reply:
Gets too verbose IMO, its like an if else,